### PR TITLE
fix: get_block_transactions_process

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

File: sync/src/relayer/compact_block_process.rs [hunk 1]
```diff
@@ -226,8 +226,6 @@ impl<'a> CompactBlockProcess<'a> {
                 }
             }
 
-            assert!(!missing_transactions.is_empty() || !missing_uncles.is_empty());
```
Comment (u2) [inline sync/src/relayer/compact_block_process.rs]: If there are no missing transactions and missing uncles, it should return `Ok(Status::AcceptBlock);` https://github.com/nervosnetwork/ckb/pull/1680/files#diff-98eccb89a5611099d6adca819c9c8b08L202

Comment (zhangsoledad) [reply sync/src/relayer/compact_block_process.rs]:
```

Comment (u2) [reply sync/src/relayer/compact_block_process.rs]: If this is a collision, the `missing_transactions` is not empty.

**Comment by quake** [comment]
bors r=u2,quake

**Comment by zhangsoledad** [comment]
bors retry

**Pull request merged**